calculator.rb

```diff
@@ -0,0 +1,108 @@
+
+#ask the user for an operation and two numbers
+puts "Choose your operator: \nOptions \n+\n-\n*\n/\nadd\nsubtract\nmultiply\ndivide\n"
+operator = gets.chomp.downcase
+
+#array of valid operators
+valid_operators = ["+", "-", "*", "/", "add", "subtract", "multiply", "divide"]
+
+#check if chosen operator is valid
+until valid_operators.include? operator
+  puts "Please enter a valid operator:"
+  operator = gets.chomp.downcase
+end
+
+#method that checks if numeric values are valid
+class String
+  def valid_float?
+    Float(self) rescue false
+  end
+end
+
+#input for 2 numeric values
+puts "Please enter your first numeric value:"
+value1 = gets.chomp
+until value1.valid_float?
+  puts "Please enter a valid numeric value:"
+  value1 = gets.chomp
+end
+value1 = value1.to_f
+
+puts "Please enter your second numeric value:"
+value2 = gets.chomp
+until value2.valid_float?
+  puts "Please enter a valid numeric value:"
+  value2 = gets.chomp
+end
+value2 = value2.to_f
+
+#chooses operation based on operator
+case operator
+  when "+", "add"
+    answer = value1 + value2
+  when "-", "subtract"
+    answer = value1 - value2
+  when "*", "multiply"
+    answer = value1 * value2
+  when "/", "divide"
+#checks for zeros as second value
+#note: this doesn't also check if it's a valid float :(
+    until value2 != 0
+      puts "Please enter a valid value that isn't 0:"
+      value2 = gets.chomp.to_f
```

Comment:
Just a small note... in this flow, your code doesn't check if the new number that replaces 0 in "divide" operations using your `valid_float?` method. It doesn't matter too much in the end because `.to_f` actually handles it pretty well, but I just wanted to let you know!